### PR TITLE
Fix various coverity errors

### PR DESCRIPTION
**Pull request by janjust**

### CODE REVIEW

**Comment by jsquyres** [comment]
ok to test

**Comment by jsquyres** [comment]
@janjust Can you add a signoff line on your commit? https://github.com/open-mpi/ompi/blob/master/.github/CONTRIBUTING.md#contributors-declaration

Thanks!

File: opal/mca/pmix/pmix4x/pmix/include/pmix_common.h.in [hunk 1]
```diff
@@ -1538,6 +1538,9 @@ typedef struct pmix_info {
     do {                                                        \
         pmix_info_t *_i;                                        \
         (m) = (pmix_info_t*)calloc((n), sizeof(pmix_info_t));   \
+        if ( NULL == (m) ) {                                    \
```
Comment (rhc54) [inline opal/mca/pmix/pmix4x/pmix/include/pmix_common.h.in]: There isn't any loop here, so you cannot "break" out.

**Comment by janjust** [comment]
@rhc54 done, remove pmix files from this commit
@jsquyres add signoff line in commit

File: opal/mca/common/ucx/common_ucx_wpool.h [hunk 2]
```diff
@@ -441,7 +441,7 @@ opal_common_ucx_wpmem_post(opal_common_ucx_wpmem_t *mem, ucp_atomic_post_op_t op
                              len, rem_addr, rkey);
     if (OPAL_UNLIKELY(status != UCS_OK)) {
         MCA_COMMON_UCX_ERROR("ucp_atomic_post failed: %d", status);
-        rc = OPAL_ERROR;
+        return OPAL_ERROR;
```
Comment (artpol84) [inline opal/mca/common/ucx/common_ucx_wpool.h]: @janjust you can't return here without releasing the lock in the line 439

File: opal/mca/common/ucx/common_ucx_wpool.h [hunk 3]
```diff
@@ -480,7 +480,7 @@ opal_common_ucx_wpmem_fetch(opal_common_ucx_wpmem_t *mem,
                                           winfo->worker);
     if (OPAL_UNLIKELY(status != UCS_OK)) {
         MCA_COMMON_UCX_ERROR("ucp_atomic_cswap64 failed: %d", status);
-        rc = OPAL_ERROR;
+        return OPAL_ERROR;
```
Comment (artpol84) [inline opal/mca/common/ucx/common_ucx_wpool.h]: Same issue here

File: opal/mca/common/ucx/common_ucx_wpool.h [hunk 4]
```diff
@@ -364,7 +364,7 @@ opal_common_ucx_wpmem_putget(opal_common_ucx_wpmem_t *mem, opal_common_ucx_op_t
 
     if (OPAL_UNLIKELY(status != UCS_OK && status != UCS_INPROGRESS)) {
         MCA_COMMON_UCX_ERROR("%s failed: %d", called_func, status);
-        rc = OPAL_ERROR;
+        return OPAL_ERROR;
```
Comment (artpol84) [inline opal/mca/common/ucx/common_ucx_wpool.h]: Same here

File: ompi/mca/osc/ucx/osc_ucx_active_target.c [hunk 5]
```diff
@@ -256,6 +256,9 @@ int ompi_osc_ucx_post(struct ompi_group_t *group, int assert, struct ompi_win_t
                                                  myrank + 1, &result, sizeof(result),
                                                  remote_addr);
                 if (ret != OMPI_SUCCESS) {
+                    free(ranks_in_grp);
+                    free(ranks_in_win_grp);
+                    ompi_group_free(&win_group);
```
Comment (artpol84) [inline ompi/mca/osc/ucx/osc_ucx_active_target.c]: Is this a verification path?
Does ompi_comm_group allocate a group? 
Just to make sure that it isn't a pointer to an internal object.

Comment (janjust) [reply ompi/mca/osc/ucx/osc_ucx_active_target.c]: yes, it allocates it in line 226, it's released at proper function exit, line 284

File: opal/mca/common/ucx/common_ucx_wpool.h (hunk 2 above)
Comment (janjust) [reply opal/mca/common/ucx/common_ucx_wpool.h]: good catch, fixed in forced push, for all 3 cases

**Comment by janjust** [comment]
@artpol84 , addressed your comments in latest force push

**Pull request closed without merging**